javalin/src/main/java/io/javalin/jetty/JavalinJettyServlet.kt

```diff
@@ -68,9 +70,20 @@ class JavalinJettyServlet(val cfg: JavalinConfig, private val httpServlet: Javal
             pathParamMap = entry.extractPathParams(requestUri),
         )
         if (!allowedByAccessManager(entry, upgradeContext)) return res.sendError(HttpStatus.UNAUTHORIZED.code, HttpStatus.UNAUTHORIZED.message)
+        val beforeUpgradeHandlers = cfg.pvt.internalRouter.findHttpHandlerEntries(HandlerType.WEBSOCKET_BEFORE_UPGRADE, requestUri)
+        val afterUpgradeHandlers = cfg.pvt.internalRouter.findHttpHandlerEntries(HandlerType.WEBSOCKET_AFTER_UPGRADE, requestUri)
+
         req.setAttribute(upgradeContextKey, upgradeContext)
         setWsProtocolHeader(req, res)
-        super.service(req, res) // everything is okay, perform websocket upgrade
+
+        try {
+            beforeUpgradeHandlers.forEach { it.handle(upgradeContext, requestUri) }
+            super.service(req, res) // everything is okay, perform websocket upgrade
+            afterUpgradeHandlers.forEach { it.handle(upgradeContext, requestUri) }
+        } catch (e: Exception) {
+            if (e is EndOfLifecycleException) return
+            cfg.pvt.internalRouter.handleHttpException(upgradeContext, e)
+        }
```

Comment:
hmm, will try to take a look so it's integrated with the task queue, but that is a bigger refactor which should probably happen in another PR.

Reply:
It shouldn't be that hard, it's just about wrapping those matched handlers into the `io.javalin.http.servlet.Task`. Something like:

```kotlin
// add before handlers
 val beforeUpgradeHandlers = cfg.pvt.internalRouter
  .findHttpHandlerEntries(HandlerType.WEBSOCKET_BEFORE_UPGRADE, requestUri)
  .forEach { handler -> upgradeContext.tasks.offer(Task { handler.execute([...]) } }}
// add service handler
// after handlers

while (upgradeContext.tasks.isNotEmpty()) {
 val task = upgradeContext.tasks.poll()
 task.execute()
} 
```

a little bit of pseudocode, because I don't remember the API.

Reply:
Your pseudo code was exactly what I needed to use the task queue api. I switched over with the most recent commit and all tests are still green. :)